### DISCUSSION TITLE
Migrator does not use correct logger

### DISCUSSION THREAD

**Discussion by JulianCissen** [mikro-orm/mikro-orm#6039] [General]
### Describe the bug

Migrator does not use the connection's loggerFactory when defined.

### Reproduction

Steps to reproduce:
1. Generate a migration.
2. Run the code below.

Expected result:
`{ "foo": "bar" }` is logged.

Actual result:
The default logging is logged.

I think this is due to [packages/migrations/src/Migrator.ts ](https://github.com/mikro-orm/mikro-orm/blob/44e921b3fcb9cda54139ffae31b3a27eafb33c78/packages/migrations/src/Migrator.ts#L140) not using the `getLogger` method in stead.

```ts
import { Entity, PrimaryKey } from "@mikro-orm/core";
import { Migrator } from "@mikro-orm/migrations";
import { MikroORM } from "@mikro-orm/postgresql";
import { v4 } from "uuid";

@Entity()
export class Author {
    @PrimaryKey()
    id = v4();
}

const start = async () => {
    const db = await MikroORM.init({
        entities: [Author],
        dbName: "debug",
        host: "localhost",
        port: 5432,
        user: "postgres",
        password: "postgres",
        //logger: (message) => console.log({ foo: 'bar' }),
        loggerFactory: (options) => new CustomMikroOrmLogger(options),
        extensions: [Migrator],
    });

    const migrator = db.getMigrator();
    await migrator.up();
};

start();

import {
    type LogContext,
    type LoggerNamespace,
    type LoggerOptions,
    SimpleLogger,
} from "@mikro-orm/core";

export class CustomMikroOrmLogger extends SimpleLogger {
    /**
     * @inheritdoc
     */
    public override log(
        namespace: LoggerNamespace,
        message: string,
        context?: LogContext
    ): void {
        if (!this.isEnabled(namespace, context)) {
            return;
        }

        console.log({ foo: "bar" });
    }

    public static override create(
        options: LoggerOptions
    ): CustomMikroOrmLogger {
        return new CustomMikroOrmLogger(options);
    }
}
```

I'm sorry for not providing the repro in the requested format, I used my own sandbox that I already used before the format was prescribed. If this is required I may be able to provide it in the weekend. I'll make sure to use the prescribed format next time.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.10

### Node.js version

18.20.4

### Operating system

Windows

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Migrations as well as schema generator are using different loggers (or should I say different logging options, the underlying logger should be the same), because they would flood the logs easily, there is a lot of queries under the hood to check the current state of the database etc. You can enable verbose logging to see those. 

This is in fact about the whole CLI, not really about the migrator - if you would use that programmatically, it should be logging stuff as usual, but the CLI is always disabling the logging unless you enable verbose logging:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/cli/src/CLIHelper.ts#L24-L25

I am open for suggestions how to make this cleaner.

**Comment by JulianCissen** [comment]
I'm not sure I follow.

I am using the migrator programmatically in the example, and it is not logging like the 'normal' logs (let's say when I just run a find query and have debug logging on). The normal logs in this case use the loggerFactory, the migrator doesn't.

My use case here is that in my production environment, the logging aggregator relies on the log output being JSON.

When the migrator is used programmatically, I see just two log entries per migration; processing and applied. How would the logs be flooded if it uses the loggerFactory? I assume that all the queries used to check the database are logged only if debug is set to true or includes 'query'. But if I set up my configuration like that, that's the output I would expect. I don't really see a problem here. Please help me understand your point better.

**Reply by B4nan** [reply]
Oh, now I get what you mean. It should be `this.config.getLogger()` instead of `this.config.get('logger')` in there.

**Reply by JulianCissen** [reply]
Yeah, that was my assumption in the initial post as well.

**Reply by B4nan** [reply]
e845c26b74a85948249e72dc75817a58e0cf7fd6

**Reply by alexandre-abrioux** [reply]
Thanks for this @B4nan! This is what I attempted to tackle back in https://github.com/mikro-orm/mikro-orm/pull/4155. It fixes my issue too 🙂
